server/transaction/manager.go

```diff
@@ -59,20 +59,44 @@ type Tx interface {
 	Rollback(ctx context.Context) error
 }
 
+type stagedDatabase interface {
+	IsMetadataChange() bool
+}
+
 // SessionCtx is used to store any baggage for the lifetime of the transaction. We use it to stage the database inside
 // a transaction when the transaction is performing any DDLs.
 type SessionCtx struct {
-	db any
+	db                    stagedDatabase
+	noMetadataChanged     *bool
+	metadataChangeSession bool
```

Comment:
Why we need two variables?

Reply:
`metadataChangeSession` is whether the session is requesting metadata change, I have added it here so that logic of checking whether we need to increment version or not will be at one place.